### PR TITLE
MM-18901 - Remove unused status flags in requests/preferences.ts #12521

### PR DESCRIPTION
**Pull request by atulya-pandey**
#### Summary

Removal of unused status flags from requests/preferences.ts

- MY_PREFERENCES_REQUEST
- MY_PREFERENCES_SUCCESS
- MY_PREFERENCES_FAILURE
- SAVE_PREFERENCES_REQUEST
- SAVE_PREFERENCES_SUCCESS
- SAVE_PREFERENCES_FAILURE
- DELETE_PREFERENCES_REQUEST
- DELETE_PREFERENCES_SUCCESS
- DELETE_PREFERENCES_FAILURE

#### Ticket Link

[MM-18901](https://mattermost.atlassian.net/browse/MM-18901)
Fixes [#12521](https://github.com/mattermost/mattermost-server/issues/12521)
#### Checklist

- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @atulya-pandey for the contribution! :tada: Heads up that reviews may be slower with the holidays

**Comment by jasonblais** [comment]
@sudheerDev / @saturninoabril / @bradjcoughlin quick reminder to help with reviews when you have a chance, thanks! :)

File: src/types/store.ts [hunk 1]
```diff
@@ -63,7 +63,6 @@ export type GlobalState = {
         posts: PostsRequestsStatuses;
         teams: TeamsRequestsStatuses;
         users: UsersRequestsStatuses;
-        preferences: PreferencesRequestsStatuses;
```
Comment (sudheerDev) [inline src/types/store.ts]: Can you remove the corresponding 

```
export type PreferencesRequestsStatuses = {
    getMyPreferences: RequestStatusType;
    savePreferences: RequestStatusType;
    deletePreferences: RequestStatusType;
};
```
In src/types/requests.ts

Comment (atulya-pandey) [reply src/types/store.ts]: @sudheerDev  Done :+1:

Comment (atulya-pandey) [reply src/types/store.ts]: Hi @sudheerDev , please let me know if there are any changes required.

Comment (sudheerDev) [reply src/types/store.ts]: @atulya-pandey thanks 👍

**Comment by atulya-pandey** [comment]
Hi @sudheerDev , I mistakenly clicked on the '**Request re-review**'. Can you please approve the PR again? Apologies for the inconvenience.

**Comment by sudheerDev** [comment]
@atulya-pandey No problem at all

File: src/store/initial_state.ts [hunk 2]
```diff
@@ -220,20 +220,6 @@ const state: GlobalState = {
...
-            },
-            savePreferences: {
-                status: 'not_started',
-                error: null,
-            },
-            deletePreferences: {
-                status: 'not_started',
-                error: null,
-            },
-        },
```
Comment (saturninoabril) [inline src/store/initial_state.ts]: The corresponding [state.requests](https://github.com/mattermost/mattermost-mobile/blob/master/app/initial_state.js#L127-L140) on mobile should be removed too one the mattermost-redux hash is updated.

**Comment by saturninoabril** [comment]
Thanks @atulya-pandey!

File: src/store/initial_state.ts (hunk 2 above)
Comment (saturninoabril) [reply src/store/initial_state.ts]: Follow up PR for mobile - https://github.com/mattermost/mattermost-mobile/pull/3836

**Pull request merged**